src/webgpu/api/validation/encoding/cmds/render/draw.spec.ts

```diff
@@ -7,8 +7,120 @@ and parameters as expect.
 import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
 import { ValidationTest } from '../../../validation_test.js';
 
+// Class that indicate how to call a draw function
+class DrawCall {
+  test: ValidationTest;
+  drawType: 'draw' | 'drawIndexed' | 'drawIndirect' | 'drawIndexedIndirect' = 'draw';
+
+  // Draw
+  vertexCount: number = 4;
+  firstVertex: number = 0;
+
+  // DrawIndexed
+  indexCount: number = 4;
+  firstIndex: number = 0;
+  baseVertex: number = 0;
+
+  // Both Draw and DrawIndexed
+  instanceCount: number = 1;
+  firstInstance: number = 0;
+
+  constructor(test: ValidationTest) {
+    this.test = test;
+  }
+
+  callDraw(encoder: GPURenderEncoderBase) {
+    switch (this.drawType) {
+      case 'draw': {
+        encoder.draw(this.vertexCount, this.instanceCount, this.firstVertex, this.firstInstance);
+        break;
+      }
+      case 'drawIndexed': {
+        encoder.drawIndexed(
+          this.indexCount,
+          this.instanceCount,
+          this.firstIndex,
+          this.baseVertex,
+          this.firstInstance
+        );
+        break;
+      }
+      case 'drawIndirect': {
+        encoder.drawIndirect(this.generateIndirectBuffer(), 0);
+        break;
+      }
+      case 'drawIndexedIndirect': {
+        encoder.drawIndexedIndirect(this.generateIndexedIndirectBuffer(), 0);
+        break;
+      }
+    }
+  }
+
+  // Create an indirect buffer containing draw call values
+  private generateIndirectBuffer(): GPUBuffer {
+    const indirectArray = new Int32Array([
+      this.vertexCount,
+      this.instanceCount,
+      this.firstVertex,
+      this.firstInstance,
+    ]);
+    return this.test.makeBufferWithContents(indirectArray, GPUBufferUsage.INDIRECT);
+  }
+
+  // Create an indirect buffer containing indexed draw call values
+  private generateIndexedIndirectBuffer(): GPUBuffer {
+    const indirectArray = new Int32Array([
+      this.indexCount,
+      this.instanceCount,
+      this.firstIndex,
+      this.baseVertex,
+      this.firstInstance,
+    ]);
+    return this.test.makeBufferWithContents(indirectArray, GPUBufferUsage.INDIRECT);
+  }
+}
+
 class F extends ValidationTest {
-  // TODO: Implement the helper functions
+  // Create a GPUBuffer that suitable for given index count and format
```

Comment:
For documentation comments please use tsdoc/jsdoc syntax as in other files: `/** blah blah */`